docs/using.md

```diff
@@ -494,6 +541,73 @@ spec:
           ${inputs.resources.testCluster.Name} apply -f /workspace/service.yaml'
 ```
 
+### Storage Resource
+
+Storage resource represents a blob storage, that contains either an object or directory.
+Adding the storage resource as an input to a task will download the blob and allow the task to perform the required actions on the contents of the blob. Blob storage type "Google Cloud Storage"(gcs) is supported as of now.
+
+#### GCS Storage Resource
+
+GCS Storage resource points to "Google Cloud Storage" blob.
+
+To create a GCS type of storage resource using the `PipelineResource` CRD:
+
+```yaml
+apiVersion: pipeline.knative.dev/v1alpha1
+kind: PipelineResource
+metadata:
+  name: wizzbang-storage
+  namespace: default
+spec:
+  type: storage
+  params:
+    - name: type
+      value: gcs
+    - name: location
+      value: gs://some-bucket
+```
+
+Params that can be added are the following:
+
+1. `location`: represents the location of the blob storage.
+1. `type`: represents the type of blob storage. Currently there is implementation for only `gcs`.
+1. `dir`: represents whether the blob storage is a directory or not.
+    1. If artifact is a directory then `-r`(recursive) flag is used to copy all files under source directory to gcs bucket. Eg: `gsutil cp -r source_dir gs://some-bucket`
```

Comment:
is there a default value for the `dir` field? since its not the example here, its not clear what the value would be